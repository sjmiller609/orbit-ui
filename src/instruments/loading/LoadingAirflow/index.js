'use strict'
import React from 'react'
import { Icon } from 'instruments'

import s from './styles.scss'

class LoadingAirflow extends React.Component {
  render() {
    return (
      <div className={s.container}>
        <span className={s.logo}>{<Icon icon="airflow_astro" />}</span>
      </div>
    )
  }
}

export default LoadingAirflow
