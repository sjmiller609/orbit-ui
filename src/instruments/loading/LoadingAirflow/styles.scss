@import 'src/styles/mixins.scss';

@keyframes spin {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}

.container {
  display: flex;
  margin: 5rem auto;
}

.logo {
  animation: spin 1.5s linear infinite;

  img {
    width: 100px;
    height: 100px;
  }
}
