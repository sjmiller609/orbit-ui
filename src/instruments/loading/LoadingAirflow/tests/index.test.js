import React from 'react'
import { shallow } from 'enzyme'
import LoadingAirflow from '../index'

describe('LoadingAirflow', () => {
  it('should be defined', () => {
    expect(LoadingAirflow).toBeDefined()
  })

  it('should render correctly', () => {
    const wrapper = shallow(<LoadingAirflow />)
    expect(wrapper).toMatchSnapshot()
  })
})
