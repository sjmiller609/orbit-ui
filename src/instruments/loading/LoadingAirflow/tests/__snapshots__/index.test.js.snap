// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingAirflow should render correctly 1`] = `
<div
  className="container"
>
  <span
    className="logo"
  >
    <Icon
      icon="airflow_astro"
    />
  </span>
</div>
`;
